### DISCUSSION TITLE
Reduce initial server response time, speed index, minimize main-thread work

### DISCUSSION THREAD

**Discussion by francisrod01** [vercel/next.js#14301] [Help]
An unanswered related post https://github.com/vercel/next.js/discussions/13763

What can I reduce server response times using SSR + `getInitialProps` / `getServerSideProps` to fetch data?
It's for a better UX, speed index and load performance.
I think it's something that everyone using Next.js want to know about.

![image](https://user-images.githubusercontent.com/4691768/84977605-23af5c00-b101-11ea-8f03-a053987b715c.png)

Environment:

- next: `9.3.6` 
- react: `16.13.1` 
- react-dom: `16.13.1`
- @material-ui/core: `4.9.12`
- @material-ui/lab: `4.0.0-alpha.51`
- node: `10.20.1`
- npm: `6.14.4`
- OS: `Linux debian 4.9.0-12-amd64`

Recently Next.js lanched `getStaticProps` and `getServerSideProps` and I noticed that it's recommended using `getStaticProps` instead of others for SEO.

[nextjs.org/docs - data fetching, when-should I use getStaticProps](https://nextjs.org/docs/basic-features/data-fetching#when-should-i-use-getstaticprops)

**The main issues are described in the screenshots below:**

![image](https://user-images.githubusercontent.com/4691768/84976364-12b11b80-b0fe-11ea-884a-189498bfa35b.png)

![image](https://user-images.githubusercontent.com/4691768/84977674-535e6400-b101-11ea-90b9-3b439b0ebf9b.png)

**Old discussions:**
- https://twitter.com/francisrod01/status/1217571749077889031
- [@spectrum.chat/next-js-and-google-pagespeed-optimisation](https://spectrum.chat/next-js/general/next-js-and-google-pagespeed-optimisation~e81cebf4-18c1-4938-a106-0bc0c29c33ce)

**Comment by dukesx** [comment]
Use Next Bundle Analyzer for splitting code into maximum amount of bundles. See next.config.js in "with-Ant-Design" example. You should find the proper code. Also, Use lazy loading libraries or SWR to lazy load data and try to serve images in Webp format. You should be fine after that.

Note: I am using Ant Design and I have the same stats. Pity I was thinking to switch to material UI because I thought it would be more optimised but guess not. You should know that UI frameworks like Material UI and Ant Design are COMPLETE set of components with very heavy CSS, so don't expect much performance benifits. Just enjoy that you didn't have to write all that Frontend code for those components, animations, styling, responsiveness etc. Trust me, it's a nightmare to do it on your own.

**Reply by francisrod01** [reply]
Hey @dukesx, thanks for reaching me out.
You got a great and useful comment here and I want to let you know that I'm reading about those topics right now on [web.dev](https://web.dev/).

**Reply by dukesx** [reply]
You are most welcome and apologies for the late reply, i am quite busy with a blog project i am working on nowadays (this is where i am using Ant design).
